src/linux_mcp_server/tools/storage.py

```diff
@@ -83,287 +99,157 @@ async def list_block_devices(
         return f"Error listing block devices: {str(e)}"
 
 
-@mcp.tool()
+@mcp.tool(
+    title="List directories",
+    description="List directories under a specified path with flexible sorting options.",
+    annotations=ToolAnnotations(readOnlyHint=True),
+)
 @log_tool_call
-async def list_directories_by_size(  # noqa: C901
-    path: str,
-    top_n: int | float,
-    host: str | None = None,
-    username: str | None = None,
-) -> str:
-    """
-    List directories under a specified path sorted by size (largest first).
-
-    This function uses efficient Linux primitives (du command) to calculate directory
-    sizes, making it much faster than Python-based directory traversal.
-
-    Args:
-        path: The directory path to analyze
-        top_n: Number of top directories to return (1-1000). Accepts int or float
-               (floats are truncated to integers)
-        host: Optional remote host to connect to
-        username: Optional SSH username (required if host is provided)
-
-    Returns:
-        Formatted string with directory sizes, or error message if validation fails
-
-    Security Features:
-        - Path validation and resolution using pathlib
-        - Command parameters passed as list (not shell string)
-        - Input sanitization for all parameters
-        - Graceful error handling for permission issues
-    """
-    import os
-
+async def list_directories(  # noqa: C901
+    path: t.Annotated[str, Field(description="The directory path to analyze")],
+    order_by: t.Annotated[
+        str, Field(description="Sort order - 'size', 'name', or 'modified' (default: 'name')")
+    ] = "name",
+    sort: t.Annotated[
+        str, Field(description="Sort direction - 'ascending' or 'descending' (default: 'ascending')")
+    ] = "ascending",
+    top_n: t.Annotated[
+        int | None,
+        Field(description="Optional limit on number of directories to return (1-1000, only used with size ordering)"),
+    ] = None,
```

Comment:
Rather than validating this value later using a custom class, use Pydantic's validation machinery.
```suggestion
    top_n: t.Annotated[
        int | None,
        Field(description="Optional limit on number of directories to return (1-1000, only used with size ordering)", gt=0, lt=1_000),
    ] = None,
```